Jenkinsfile_k8s

```diff
@@ -1,3 +1,39 @@
+def recordDeployment(owner, repo, status, environmentURL, environment = "preview", description = "Deploy to preview environment") {
+  def deployURL = "https://api.github.com/repos/${owner}/${repo}/deployments"
+  def response = httpRequest(
+    authentication: 'github-app-infra',
+    httpMode: 'POST',
+    requestBody: writeJSON(returnText: true, json: [
+      "ref": env.GIT_COMMIT,
+      "environment": environment,
+      "description": description
+    ]),
+    responseHandle: 'STRING',
+    url: deployURL,
+  )
+
+  def id = readJSON(text: response.content).id
+  if (id == ""){
+    error("Unable to create deployment")
+  }
+
+  httpRequest(
+    authentication: 'github-app-infra',
+    httpMode: 'POST',
+    requestBody: writeJSON(
+      returnText: true,
+      json: [
+        "state": status,
+        "environment": environment,
+        "description": description,
+        "log_url": "${env.BUILD_URL}/console",
+        "environment_url": environmentURL
+      ]
+    ),
+    url: "${deployURL}/${id}/statuses"
+  )
+}
```

Comment:
This code is not really following the good practises for pipelines: https://www.jenkins.io/doc/book/pipeline/pipeline-best-practices/#making-sure-to-use-groovy-code-in-pipelines-as-glue .

WDYT avout running this as a groovy script (or whatever other language you would like) on an agent?